api/api_handler.go

```diff
@@ -111,20 +111,26 @@ func (a *ApiHandler) GetAddressStats(ctx echo.Context, address AddressFilter) er
 // GetValidators returns all validators for the selected week and the number of PBFT blocks they produced
 func (a *ApiHandler) GetValidators(ctx echo.Context, params GetValidatorsParams) error {
 	fmt.Println("GetValidators")
-	var data []Validator
+
+	stats := a.storage.GetWeekStats(int(params.Week.Year), int(params.Week.Week))
+	ret, pagination := stats.GetPaginated(int(params.Pagination.Start), params.Pagination.Limit)
+
 	response := struct {
 		DagsPaginatedResponse
-		Data []Validator
+		Data []Validator `json:"data"`
 	}{
-		Data: data,
+		DagsPaginatedResponse: *pagination,
+		Data:                  ret,
 	}
+
 	return ctx.JSON(http.StatusOK, response)
 }
 
 // GetValidatorsTotal returns total number of PBFT blocks produced in selected week
 func (a *ApiHandler) GetValidatorsTotal(ctx echo.Context, params GetValidatorsTotalParams) error {
+	stats := a.storage.GetWeekStats(int(params.Filter.Year), int(params.Filter.Week))
 	var count CountResponse
-	count.Total = 0
+	count.Total = uint64(stats.Total)
```

Comment:
@rattrap do we need this data somewhere separate from that validators list?
If not I think it makes sense to merge this 2 requests

Reply:
Yes. We have a summary at the start of the page - https://testnet.explorer.taraxa.io/node

And then the list of validators under that. "Total" for the validators list may mean total number of block for that page or total number of validators returned. At the top of the page we need the total for all validators (all pages) for the week.

Reply:
yeah, but if it is on the same page we can get all the data with 1 request

Reply:
We could have it in the validators request too and when integrating we can have to option to just use one request.